src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/service/VisitService.kt

```diff
@@ -263,6 +265,24 @@ class VisitService(
   private fun getNextEvent(): Long {
     return visitVisitorRepository.getEventId()
   }
+
+  fun getVisit(visitId: Long): VisitResponse {
+    return visitRepository.findByIdOrNull(visitId)?.run {
+      return VisitResponse(
+        visitId = this.id,
+        offenderNo = this.offenderBooking.offender.nomsId,
+        prisonId = this.location.id,
+        startDateTime = LocalDateTime.from(this.startDateTime),
+        endDateTime = LocalDateTime.from(this.endDateTime),
+        visitType = VisitResponse.CodeDescription(this.visitType.code, this.visitType.description),
+        visitStatus = VisitResponse.CodeDescription(this.visitStatus.code, this.visitStatus.description),
+        agencyInternalLocation = this.agencyInternalLocation?.let { VisitResponse.CodeDescription(it.locationCode!!, it.description!!) },
+        commentText = this.commentText,
+        visitorConcernText = this.visitorConcernText,
+        visitors = this.visitors.filter { visitor -> visitor.person != null }.map { visitor -> VisitResponse.Visitor(visitor.person!!.id, visitor.groupLeader) }
+      )
+    } ?: throw NotFoundException("visit id $visitId")
+  }
```

Comment:
Are you deliberately keeping the optional "this." ?

Reply:
yes - might be easier to discuss